server/src/com/cloud/storage/snapshot/SnapshotManagerImpl.java

```diff
@@ -1123,12 +1123,24 @@ public SnapshotInfo takeSnapshot(VolumeInfo volume) throws ResourceAllocationExc
             }
 
             SnapshotInfo snapshotOnPrimary = snapshotStrategy.takeSnapshot(snapshot);
-            if (payload.getAsyncBackup()) {
-                backupSnapshotExecutor.schedule(new BackupSnapshotTask(snapshotOnPrimary, snapshotBackupRetries - 1, snapshotStrategy), 0, TimeUnit.SECONDS);
+            boolean backupFlag = BackupSnapshotAfterTakingSnapshot.value() == null ||
```

Comment:
Could you rename this variable to something more descriptive?

Reply:
done